frontend/src/scenes/data-warehouse/external/forms/SourceForm.tsx

```diff
@@ -1,19 +1,60 @@
-import { LemonFileInput, LemonInput, LemonSelect, LemonSwitch, LemonTextArea } from '@posthog/lemon-ui'
+import { LemonDivider, LemonFileInput, LemonInput, LemonSelect, LemonSwitch, LemonTextArea } from '@posthog/lemon-ui'
 import { Form, Group } from 'kea-forms'
 import { LemonField } from 'lib/lemon-ui/LemonField'
 
 import { SourceConfig, SourceFieldConfig } from '~/types'
 
 import { SOURCE_DETAILS, sourceWizardLogic } from '../../new/sourceWizardLogic'
 import { DataWarehouseIntegrationChoice } from './DataWarehouseIntegrationChoice'
+import { parseConnectionString } from './parseConnectionString'
 
 export interface SourceFormProps {
     sourceConfig: SourceConfig
     showPrefix?: boolean
     jobInputs?: Record<string, any>
 }
 
+const CONNECTION_STRING_DEFAULT_PORT = {
+    Postgres: 5432,
+}
+
 const sourceFieldToElement = (field: SourceFieldConfig, sourceConfig: SourceConfig, lastValue?: any): JSX.Element => {
+    if (field.type === 'text' && field.name === 'connection_string') {
```

Comment:
ideally field type is just determined by "type" but fine leaving this for now and refactoring later if we have a bunch of conditionals

Reply:
👍